primitives/xcm/src/lib.rs

```diff
@@ -49,3 +58,66 @@ impl<
 		Ok((what, amount))
 	}
 }
+
+pub struct TrappistDropAssets<
+	AssetId,
+	AssetIdInfoGetter,
+	AssetsPallet,
+	BalancesPallet,
+	XcmPallet,
+	AccoundId,
+>(PhantomData<(AssetId, AssetIdInfoGetter, AssetsPallet, BalancesPallet, XcmPallet, AccoundId)>);
+impl<AssetId, AssetIdInfoGetter, AssetsPallet, BalancesPallet, XcmPallet, AccountId> DropAssets
+	for TrappistDropAssets<
+		AssetId,
+		AssetIdInfoGetter,
+		AssetsPallet,
+		BalancesPallet,
+		XcmPallet,
+		AccountId,
+	> where
+	AssetId: Clone,
+	AssetIdInfoGetter: AssetMultiLocationGetter<AssetId>,
+	AssetsPallet: Inspect<AccountId, AssetId = AssetId>,
+	BalancesPallet: Currency<AccountId>,
+	XcmPallet: DropAssets,
+{
+	fn drop_assets(origin: &MultiLocation, assets: Assets) -> u64 {
+		let multi_assets: Vec<MultiAsset> = assets.into();
+		let mut trap: Vec<MultiAsset> = Vec::new();
+
+		for asset in multi_assets {
+			if let MultiAsset { id: Concrete(location), fun: Fungible(amount) } = asset.clone() {
+				// is location a fungible on AssetRegistry?
+				if let Some(asset_id) = AssetIdInfoGetter::get_asset_id(location.clone()) {
+					let min_balance = AssetsPallet::minimum_balance(asset_id);
+
+					// only trap if amount ≥ min_balance
+					// do nothing otherwise (asset is lost)
+					if min_balance <= amount.saturated_into::<AssetsPallet::Balance>() {
+						trap.push(asset);
+					}
+
+				// is location the native token?
+				} else if location == (MultiLocation { parents: 0, interior: Here }) {
+					let min_balance = BalancesPallet::minimum_balance();
+
+					// only trap if amount ≥ min_balance
+					// do nothing otherwise (asset is lost)
+					if min_balance <= amount.saturated_into::<BalancesPallet::Balance>() {
+						trap.push(asset);
+					}
+				}
+			}
+		}
+
+		// TODO: put real weight of execution up until this point here
+		let mut weight = 0;
+
+		if !trap.is_empty() {
+			weight += XcmPallet::drop_assets(origin, trap.into());
```

Comment:
Maybe add a simple comment here so easier for newcomers (like me) to understand the reason for this: "Deposit remaining assets in holiding registry into asset trap, for later claiming via ClaimAsset instruction"

Reply:
https://github.com/paritytech/trappist/pull/109/commits/6d8df2b6377af68b4fbd2a9f32d3230ada1c8235